api/submariner/v1alpha1/servicediscovery_types.go

```diff
@@ -30,21 +30,9 @@ import (
 // ServiceDiscoverySpec defines the desired state of ServiceDiscovery
 // +k8s:openapi-gen=true
 type ServiceDiscoverySpec struct {
-	BrokerK8sApiServer       string               `json:"brokerK8sApiServer"`
-	BrokerK8sApiServerToken  string               `json:"brokerK8sApiServerToken"`
-	BrokerK8sCA              string               `json:"brokerK8sCA"`
-	BrokerK8sRemoteNamespace string               `json:"brokerK8sRemoteNamespace"`
-	ClusterID                string               `json:"clusterID"`
-	Namespace                string               `json:"namespace"`
-	Repository               string               `json:"repository,omitempty"`
-	Version                  string               `json:"version,omitempty"`
-	Debug                    bool                 `json:"debug"`
-	GlobalnetEnabled         bool                 `json:"globalnetEnabled,omitempty"`
-	BrokerK8sInsecure        bool                 `json:"brokerK8sInsecure,omitempty"`
-	CoreDNSCustomConfig      *CoreDNSCustomConfig `json:"coreDNSCustomConfig,omitempty"`
-	// +listType=set
-	CustomDomains  []string          `json:"customDomains,omitempty"`
-	ImageOverrides map[string]string `json:"imageOverrides,omitempty"`
+	BrokerDetails    BrokerDetails  `json:"broker_info"`
+	ClusterDetails   ClusterDetails `json:"cluster_info"`
```

Comment:
This is a breaking change. You can share struct components but only by “inheriting” the struct. Any changes made here shouldn’t result in changes to the CRD YAMLs.